django/db/migrations/autodetector.py

```diff
@@ -563,6 +563,19 @@ def generate_created_models(self):
                 if isinstance(base, str) and "." in base:
                     base_app_label, base_name = base.split(".", 1)
                     dependencies.append((base_app_label, base_name, None, True))
+                    # Add dependencies for base model's removed fields that are added into newly
+                    # added sub models so that field removal operations for base models can take
+                    # place before new inherited models are created.
+                    prev_state_base_model = self.from_state.models.get((base_app_label, base_name))
+                    curr_state_base_model = self.to_state.models.get((base_app_label, base_name))
+                    if prev_state_base_model and curr_state_base_model:
+                        prev_base_fields = set(name for name, field in prev_state_base_model.fields)
+                        curr_base_fields = set(name for name, field in curr_state_base_model.fields)
+                        removed_base_fields = prev_base_fields.difference(curr_base_fields)
+                        curr_model_fields = set(name for name, field in model_state.fields)
+                        removed_base_fields = removed_base_fields.intersection(curr_model_fields)
+                        for removed_base_field in removed_base_fields:
```

Comment:
This makes me wonder if we should compare field beyond its name but I can't think of a scenario where that would be problematic given we don't allow MTI field collision.